ipam/provider/aws.go

```diff
@@ -0,0 +1,258 @@
+		hostClusterName:     getValue(config.ClusterName, "CLUSTER_NAME"),
+	}
+
+	return a, nil
+}
+
+func (aws *AWS) CreateCluster(clusterName string) error {
+	if clusterName == "" {
+		return microerror.Maskf(invalidConfigError, "clusterName must not be empty")
+	}
+
+	hostAWSConfig, err := aws.hostFramework.AWSCluster(aws.hostClusterName)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = aws.installCertResources(clusterName)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deploymentName := awsConfigDeploymentName(clusterName)
+
+	o := func() error {
+		// NOTE we ignore errors here because we cannot get really useful error
+		// handling done. This here should anyway only be a quick fix until we use
+		// the helm client lib. Then error handling will be better.
+		framework.HelmCmd(fmt.Sprintf("delete --purge %s", deploymentName))
+
+		f, err := ioutil.TempFile("/tmp", deploymentName)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		defer func() {
+			// Close & remove created tempfile.
+			fName := f.Name()
+			err = f.Close()
+			if err != nil {
+				// XXX: Tempfile leak.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s'", fName))
+				return
+			}
+			err = os.Remove(fName)
+			if err != nil {
+				// XXX: Tempfile leak.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s'", fName))
+				return
+			}
+		}()
+
+		chartValuesConfig := aws.chartValuesConfig
+		chartValuesConfig.ClusterName = clusterName
+		chartValuesConfig.AWSRouteTable0 = clusterName + "_0"
+		chartValuesConfig.AWSRouteTable1 = clusterName + "_1"
+		chartValuesConfig.AWSVPCPeerID = hostAWSConfig.Spec.AWS.VPC.PeerID
+		err = aws.chartValuesTemplate.Execute(f, chartValuesConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = framework.HelmCmd(fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-aws-config-e2e-chart:stable -- -n %s --values %s", awsConfigDeploymentName(clusterName), f.Name()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(framework.ShortMaxWait, framework.ShortMaxInterval)
+	n := func(err error, delay time.Duration) {
+		log.Println("level", "debug", "message", err.Error())
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (aws *AWS) DeleteCluster(clusterName string) {
+	deploymentName := awsConfigDeploymentName(clusterName)
+	// NOTE we ignore errors here because we cannot get really useful error
+	// handling done. This here should anyway only be a quick fix until we use
+	// the helm client lib. Then error handling will be better.
+	framework.HelmCmd(fmt.Sprintf("delete --purge %s", deploymentName))
+}
+
+func (aws *AWS) installCertResources(clusterName string) error {
```

Comment:
This is pretty much a modified copy from `e2e-harness` where it was in a bit different form. I'm unsure if this should be left as-is here, or the `e2e-harness` modified to provide this feature? This test is probably the only one needing this kind of feature for now so hence I settled with copy for now.